pkg/controller/operators/olm/operator.go

```diff
@@ -1299,20 +1301,29 @@ func (a *Operator) syncOLMConfig(obj interface{}) (syncError error) {
 			return err
 		}
 
-		// Filter to unique copies
-		uniqueCopiedCSVs := map[string]struct{}{}
+		copiedCSVCount := map[string]int{}
 		for _, copiedCSV := range copiedCSVs {
-			uniqueCopiedCSVs[copiedCSV.GetName()] = struct{}{}
+			copiedCSVCount[copiedCSV.GetName()] = copiedCSVCount[copiedCSV.GetName()] + 1
 		}
 
 		csvs, err := a.lister.OperatorsV1alpha1().ClusterServiceVersionLister().ClusterServiceVersions(og.GetNamespace()).List(labels.NewSelector().Add(*nonCopiedCSVRequirement))
 		if err != nil {
 			return err
 		}
 
+		namespaces, err := a.lister.CoreV1().NamespaceLister().List(labels.Everything())
+		if err != nil {
+			return err
+		}
+
 		for _, csv := range csvs {
-			// If the correct number of copied CSVs were found, continue
-			if _, ok := uniqueCopiedCSVs[csv.GetName()]; ok == olmConfig.CopiedCSVsAreEnabled() {
+			numberOfCopiedCSVs := copiedCSVCount[csv.GetName()]
+			// Ignore NS where actual CSV is installed
+			if olmConfig.CopiedCSVsAreEnabled() && numberOfCopiedCSVs == len(namespaces)-1 {
```

Comment:
Store the value of olmConfig.CopiedCSVsAreEnabled() before iterating over the list of csvs so we don't reinitialize per loop iteration?

Reply:
Also: do we have a more robust way to identify whether the CSV we're processing actually lives in the namespace vs. making that assumption by comparing the number of copied CSVs we already calculated to the number of namespaces we expect?

Reply:
>Also: do we have a more robust way to identify whether the CSV we're processing actually lives in the namespace vs. making that assumption by comparing the number of copied CSVs we already calculated to the number of namespaces we expect?

Yeah, that's probably a nicer implementation, updated.

Reply:
> Store the value of olmConfig.CopiedCSVsAreEnabled() before iterating over the list of csvs so we don't reinitialize per loop iteration?

Good idea

Reply:
This should all be resolved now.